client/landing/gutenboarding/onboarding-block/edit.tsx

```diff
@@ -2,56 +2,88 @@
  * External dependencies
  */
 import { __ as NO__ } from '@wordpress/i18n';
-import { SelectControl } from '@wordpress/components';
+import { Button } from '@wordpress/components';
+import { map } from 'lodash';
 import { useDispatch, useSelect } from '@wordpress/data';
 import React, { useCallback } from 'react';
 
 /**
  * Internal dependencies
  */
-import { SiteType } from '../store/types';
+import { SiteType, EMPTY_FORM_VALUE } from '../store/types';
 import { STORE_KEY } from '../store';
 import './style.scss';
 
+const siteTypeOptions = {
+	[ SiteType.BLOG ]: NO__( 'with a blog.' ),
+	[ SiteType.STORE ]: NO__( 'for a store.' ),
+	[ SiteType.STORY ]: NO__( 'to write a story.' ),
+};
```

Comment:
Chatted with @razvanpapadopol and there are a few things to note here:

- Before it was inside the component, that was wasteful _on render_. That could've been improved with something like `labels = useMemo(() => ( { ...labels } ), [] );`. Probably an irrelevant, premature optimization, but worth noting.
- Moving the object literal outside of the component funciton may actually be worse. In this case, when the module is imported the code must be evaluated and the object constructed (and the translations run), whether or not the component is used and mounted. That likely translates to perf hit when we're trying to boot. Again, minor, but worth noting.

Aside from these two points, we can better leverage TypeScript for _totality_. We can either annotate the object, or we can use a typed function. They achieve similar results:

```ts
enum Type { A, B }

// 😭
const labels = {
	[Type.A]: 'It\'s of type "A"!',
	// missing Type.B - TypeScript doens't know :(
}

// Type error! 🎉
// Property '1' is missing in type '{ [Type.A]: string; }'
// but required in type 'Record<Type, string>'.
//
//    v~~~~~v
const labels_: Record<Type,string> = {
	[Type.A]: 'It\'s of type "A"!',
}

// Type error! 🎉
// Function lacks ending return statement
// and return type does not include 'undefined'.
//
//                                     v~~~~v
function getSiteTypeLabel(type: Type): string {
	switch (type) {
		case Type.A:
			return 'It\'s of type "A"!';	
	}
}
```

Of these options, given the considerations outlined earlier, I'd lean towards a top level `getSiteTypeLabel` function. It leverages TypeScript nicely for totality and defers evaluation (translate calls) to runtime.

It's worth noting that it would run string translation on every render as suggested. Again, we could reach for `useMemo` if that were a concern, but I don't believe it is.

Reply:
Thanks Jon, this is a great explanation! :star2:

One remaining question that the `getSiteTypeLabel` approach raises is how to iterate over the `enum` :thinking: (and if we _should_ iterate over an Enum, since it can theoreticall be pretty heterogenous).

Reply:
I'm going to start with the `const siteTypeOptions: Record< SiteType, string >` approach since it bypasses the enum iteration problem, doesn't change the JS, but provides the totality check :tada:  (This also means that the perf concerns that you mention remain unaddressed for now; like you say, they're hopefully not too bad, and we can address them separately.)

Reply:
(The TS error we get this way if any `SiteType` key is missing from `siteTypeOptions` is also _way_ more telling than the `lacks return statement` that we'd get from the `getSiteTypeLabel` function.)

Reply:
> I'm going to start with the `const siteTypeOptions: Record< SiteType, string >` approach 

fd5472e